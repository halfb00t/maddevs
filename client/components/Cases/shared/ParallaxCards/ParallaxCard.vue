<template>
  <div class="card-content background-color-silver">
    <img
      v-if="icon"
      v-lazy-load
      :data-src="$getMediaFromS3(`/images/Cases/${folder}/svg/${icon}.svg`)"
      :alt="alt || 'Icon'"
      :width="width"
      :height="height"
      class="card-content_icon"
    >
    <h4
      v-if="title"
      class="case_title_h4 m-10-bottom"
    >
      {{ title }}
    </h4>
    <template
      v-if="Array.isArray(descriptions)"
    >
      <TextParagraph
        v-for="(description, i) in descriptions"
        :key="description"
        :class="i === descriptions.length ? '' : 'm-24_bottom media-m-12_bottom'"
      >
        {{ description }}
      </TextParagraph>
    </template>
    <template
      v-else
    >
      <TextParagraph>
        {{ descriptions }}
      </TextParagraph>
    </template>
    <slot />
  </div>
</template>
<script>
import TextParagraph from '@/components/Cases/shared/TextParagraph'

export default {
  name: 'ParallaxCard',
  components: {
    TextParagraph,
  },

  props: {
    title: {
      type: String,
      default: '',
    },

    descriptions: {
      type: [String, Array],
      default: '',
    },

    icon: {
      type: String,
      default: '',
    },

    folder: {
      type: String,
      default: '',
    },

    alt: {
      type: String,
      default: '',
    },

    width: {
      type: Number,
      default: 0,
    },

    height: {
      type: Number,
      default: 0,
    },
  },
}
</script>
<style lang="scss" scoped>
.card-content {
  position: relative;
  overflow: hidden;
  width: auto;
  height: auto;
  padding: 32px;
  border-radius: 8px;
  &_title {
    margin-bottom: 10px;
  }

  &_icon {
    width: 51.94px;
    height: 51.94px;
    display: block;
    margin-bottom: 13px;
  }

  @media screen and (max-width: 768px) {
    &_title,
    &_icon {
      margin-bottom: 8px;
    }
  }
}
</style>
