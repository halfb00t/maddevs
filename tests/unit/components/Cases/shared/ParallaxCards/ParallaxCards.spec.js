import { render, screen } from '@testing-library/vue'
import { shallowMount } from '@vue/test-utils'
import ParallaxCards from '@/components/Cases/shared/ParallaxCards/ParallaxCards'

const props = {
  dataCards: [{
    title: 'test title',
    descriptions: 'test description',
    icon: 'test-icon',
    alt: 'test-icon',
    folder: 'testFolder',
    iconWidth: 51.94,
    iconHeight: 51.94,
  }, {
    title: 'test title 2',
    descriptions: 'test description 2',
    icon: 'test-icon-2',
    alt: 'test-icon-2',
    folder: 'testFolder',
    iconWidth: 51.94,
    iconHeight: 51.94,
  }, {
    title: 'test title 3',
    descriptions: 'test description 3',
    icon: 'test-icon-3',
    alt: 'test-icon-3',
    folder: 'testFolder',
    iconWidth: 51.94,
    iconHeight: 51.94,
  }, {
    title: 'test title 4',
    descriptions: 'test description 4',
    icon: 'test-icon-4',
    alt: 'test-icon-4',
    folder: 'testFolder',
    iconWidth: 51.94,
    iconHeight: 51.94,
  }],
  startSlice: 2,
}

const directives = {
  'lazy-load': () => {},
  'mad-parallax': () => {},
}

const mocks = {
  $getMediaFromS3: () => 'img.jpg',
}

describe('ParallaxCards component', () => {
  it('should render correctly', () => {
    const { container } = render(ParallaxCards, {
      props,
      mocks,
      directives,
    })

    expect(container).toMatchSnapshot()
  })

  it('should correctly render classname', () => {
    const { html } = render(ParallaxCards, {
      props,
      mocks,
      directives,
    })

    expect(screen.getByText(props.dataCards[0].title)).toBeTruthy()
    expect(html()).toContain('parallax_cards')
  })

  it('should return empty array from dataCards props', () => {
    const container = shallowMount(ParallaxCards, {
      props: {
        startSlice: 2,
      },
      mocks,
      directives,
    })

    expect(container.vm.$options.props.dataCards.default.call()).toEqual([])
  })
})
