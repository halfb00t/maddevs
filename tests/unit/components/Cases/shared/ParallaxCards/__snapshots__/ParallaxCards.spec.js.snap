// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParallaxCards component should render correctly 1`] = `
<div>
  <div
    class="parallax_cards"
  >
    <div
      class="parallax_cards__item"
    >
      <div
        class="card-content background-color-silver"
        iconheight="51.94"
        iconwidth="51.94"
      >
        <img
          alt="test-icon"
          class="card-content_icon"
          data-src="img.jpg"
          height="51.94"
          width="51.94"
        />
         
        <h4
          class="case_title_h4 m-10-bottom"
        >
          
    test title
  
        </h4>
         
        <p
          class="case_paragraph"
        >
          
      test description
    
        </p>
         
      </div>
      <div
        class="card-content background-color-silver"
        iconheight="51.94"
        iconwidth="51.94"
      >
        <img
          alt="test-icon-2"
          class="card-content_icon"
          data-src="img.jpg"
          height="51.94"
          width="51.94"
        />
         
        <h4
          class="case_title_h4 m-10-bottom"
        >
          
    test title 2
  
        </h4>
         
        <p
          class="case_paragraph"
        >
          
      test description 2
    
        </p>
         
      </div>
    </div>
     
    <div
      class="parallax_cards__item"
    >
      <div
        class="card-content background-color-silver"
        iconheight="51.94"
        iconwidth="51.94"
      >
        <img
          alt="test-icon-3"
          class="card-content_icon"
          data-src="img.jpg"
          height="51.94"
          width="51.94"
        />
         
        <h4
          class="case_title_h4 m-10-bottom"
        >
          
    test title 3
  
        </h4>
         
        <p
          class="case_paragraph"
        >
          
      test description 3
    
        </p>
         
      </div>
      <div
        class="card-content background-color-silver"
        iconheight="51.94"
        iconwidth="51.94"
      >
        <img
          alt="test-icon-4"
          class="card-content_icon"
          data-src="img.jpg"
          height="51.94"
          width="51.94"
        />
         
        <h4
          class="case_title_h4 m-10-bottom"
        >
          
    test title 4
  
        </h4>
         
        <p
          class="case_paragraph"
        >
          
      test description 4
    
        </p>
         
      </div>
    </div>
  </div>
</div>
`;
